packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/TeachingPopoverCarouselContext.ts

```diff
@@ -1,53 +1,32 @@
-import { createContext, useContextSelector } from '@fluentui/react-context-selector';
-import type { ContextSelector, Context } from '@fluentui/react-context-selector';
+import { CarouselContextValue } from './Carousel/useCarouselCollection';
+import { CarouselWalkerContextValue } from './Carousel/useCarouselWalker';
 import { TeachingPopoverCarouselState } from './TeachingPopoverCarousel.types';
 
-const TeachingPopoverCarouselContext: Context<TeachingPopoverCarouselContextValue> = createContext<
-  TeachingPopoverCarouselContextValue | undefined
->(undefined) as Context<TeachingPopoverCarouselContextValue>;
-
-const teachingPopoverCarouselContextDefaultValue: TeachingPopoverCarouselContextValue = {
-  currentPage: 0,
-  setCurrentPage: () => null,
-  totalPages: 1, //'One-pager' default state until proven otherwise
-  onPageChange: () => null,
-};
-
-export const TeachingPopoverCarouselProvider = TeachingPopoverCarouselContext.Provider;
-
 /**
  * Context shared between TeachingPopoverCarousel and its children components
  */
 export type TeachingPopoverCarouselContextValues = {
-  carousel: TeachingPopoverCarouselContextValue;
+  carousel: CarouselContextValue;
+  carouselWalker: CarouselWalkerContextValue;
 };
 
-/**
- * Context shared between TeachingPopoverCarousel and its children components
- */
-export type TeachingPopoverCarouselContextValue = Pick<
-  TeachingPopoverCarouselState,
-  'currentPage' | 'setCurrentPage' | 'totalPages' | 'onPageChange'
->;
-
-export const useTeachingPopoverCarouselContext_unstable = <T>(
-  selector: ContextSelector<TeachingPopoverCarouselContextValue, T>,
-): T =>
-  useContextSelector(TeachingPopoverCarouselContext, (ctx = teachingPopoverCarouselContextDefaultValue) =>
-    selector(ctx),
-  );
-
 export function useTeachingPopoverCarouselContextValues_unstable(
   state: TeachingPopoverCarouselState,
 ): TeachingPopoverCarouselContextValues {
-  const { currentPage, setCurrentPage, totalPages, onPageChange } = state;
+  const { store, value, setValue, setIndex, onPageChange, totalPages, currentIndex, find, nextPage, prevPage, active } =
```

Comment:
The walker return of `useCarouselWalker` is memoized so that it can be used in a normal react context without memoizing. Right now because you create a new object, it will get created on each re-render and trigger child re-renders. So here you should either:

1. just pass the walker return directly to the context
2. memoize a new object with all the walker functions

Option 1 is preferrable since it would avoid an extra memo